### PR TITLE
Fix the length of b Vector for all the objects and the corresponding tests

### PR DESCRIPTION
**Pull request by sylviemonet**
**Context:** Because of the tests were `np.allclose(b, 0)`, we haven't tested the length of the b vector just the values. Hence we have mistakes in the length of b Vector of rotation gate, thermal state and attenuator/amplifier channels.
 In this PR, we fix the b vectors of them and fix the tests with respect to the shape of the vector it should be. 
We also fix the type hint for the functions.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: tests/test_physics/test_triples.py [hunk 1]
```diff
@@ -48,39 +48,39 @@ def test_vacuum_state_Abc(self, n_modes):
         A, b, c = triples.vacuum_state_Abc(n_modes)
 
         assert np.allclose(A, np.zeros((n_modes, n_modes)))
-        assert np.allclose(b, np.zeros((n_modes)))
+        assert np.allclose(b, np.zeros(n_modes))
```
Comment (SamFerracin) [inline tests/test_physics/test_triples.py:51]: I think it'd be best to use `np.all` instead of `np.allclose`

Comment (sylviemonet) [reply tests/test_physics/test_triples.py:51]: Working on it

**Review by SamFerracin — APPROVED** [review]
:top:

File: mrmustard/math/backend_numpy.py [hunk 2]
```diff
@@ -59,6 +59,13 @@ def __repr__(self) -> str:
     def abs(self, array: np.ndarray) -> np.ndarray:
         return np.abs(array)
 
+    def allclose(self, array1: np.array, array2: np.array, atol: float) -> bool:
```
Comment (ziofil) [inline mrmustard/math/backend_numpy.py:62]: why is it necessary to reproduce this in the backend? There's no need for gradients

File: mrmustard/math/backend_tensorflow.py [hunk 3]
```diff
@@ -83,13 +90,13 @@ def astensor(self, array: Union[np.ndarray, tf.Tensor], dtype=None) -> tf.Tensor
         return tf.convert_to_tensor(array, dtype)
 
     def atleast_1d(self, array: tf.Tensor, dtype=None) -> tf.Tensor:
-        return tf.experimental.numpy.atleast_1d(self.astensor(array, dtype))
+        return tf.experimental.numpy.atleast_1d(self.cast(self.astensor(array), dtype))
```
Comment (ziofil) [inline mrmustard/math/backend_tensorflow.py:93]: why this change?

File: mrmustard/math/backend_numpy.py (hunk 2 above)
Comment (SamFerracin) [reply mrmustard/math/backend_numpy.py:62]: For the tests, we needed an allclose that
- takes `atol`
- checks the shapes
Numpy's and tensorflow's don't do that. So we thought it best to add it to the backend, so we can use the same `allclose` in all the test files

**Review by ziofil — APPROVED** [review]
ok. I don't get why we need to maintain our own `allclose`...

File: mrmustard/math/backend_tensorflow.py (hunk 3 above)
Comment (SamFerracin) [reply mrmustard/math/backend_tensorflow.py:93]: An error showed up that can be reproduced by the following snippet:

```
import tensorflow as tf

my_tensor = tf.convert_to_tensor([1, 2])
tf.convert_to_tensor(my_tensor, tf.complex128)
--> ValueError: Tensor conversion requested dtype complex128 for Tensor with dtype int32: <tf.Tensor: shape=(2,), dtype=int32, numpy=array([1, 2], dtype=int32)>
```
